api/handlers.go

```diff
@@ -934,6 +1057,33 @@ func (si *ServerImplementation) fetchApplications(ctx context.Context, params id
 	return apps, round, nil
 }
 
+// fetchApplications fetches all results
+func (si *ServerImplementation) fetchApplicationBoxes(ctx context.Context, params idb.ApplicationBoxQuery) (appid generated.ApplicationId, boxes []generated.Box, round uint64, err error) {
+	boxes = make([]generated.Box, 0)
+
+	err = callWithTimeout(ctx, si.log, si.timeout, func(ctx context.Context) error {
+		var results <-chan idb.ApplicationBoxRow
+		results, round = si.db.ApplicationBoxes(ctx, params)
+
+		for result := range results {
+			if result.Error != nil {
+				return result.Error
+			}
+			if appid == 0 {
+				appid = generated.ApplicationId(result.App)
+			}
+			boxes = append(boxes, result.Box)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return
+	}
+
+	return
```

Comment:
@tzaffi Minor:  Does the _suggestion_ match the PR's behavior?

If so, is there a reason to retain the extra branching?  If not, can you help me understand the difference?

```suggestion
	return
```

Reply:
Possibly the intent is to set some error specific values like the surrounding functions.  If so, then disregard the suggestion and I'll look for a commit supplying error specific values.

Reply:
Goot catch.